### PR TITLE
Cythonized version 

### PR DESCRIPTION
**Pull request by ChiCheng45**
I have a suggested change that you might want to make or not. I understand if you want to keep this repo pure python etc. But it might be good idea to just work on one repos instead of both the C and python repos.

Anyway, by using some of your C code from https://github.com/oziogos/STO-proj-AOM-overlap I've rewritten some part of pyAOMlite so that its more efficient. Basically some cython is used to wrap the C code. I also moved some other functions to cython and optimised so slower part of the code. I mainly focussed on speeding up the overlap. Also I made calculate_overlap_S_matrix only calculate part of the matrix and symmetrized it to speed it up a bit.

building is extremely easy its just 

`python setup.py build_ext --inplace`

according to my numbers this cythonized version is now only 1.5/2x slower that pure C which is fine even for my case.

projection
Total tests: 79; passed: 79/79; failed 0/79
Execution time: 105.03s

overlap
Total tests: 79; passed: 79/79; failed 0/79
Total dimers: 921
Execution time: 47.14s

### CODE REVIEW

File: src/general.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ Orestis George Ziogos, UCL
+ o.ziogos@ucl.ac.uk
+
+ For more information, examine the README file in the parent directory.
+
+-------------------------------------------------------------------------------- */
+#define cmax_length 1000
+#define sub_length 200
+#define AngToBohr 1.8897259886
+#define equal_diff 1e-6
```
Comment (oziogos) [inline src/general.h:15]: Only 
`#define equal_diff 1e-6
`
and
`#include<math.h>
`
are actually used by the embedded C functions.

Try dropping all other defines, includes, and the struct definitions.

**Review by oziogos — CHANGES_REQUESTED** [review]
Revise general.h, making sure there're no implicit declaration warnings.

**Comment by oziogos** [comment]
Nice work! Since the project got a little bit more complicated, we should compile a requirements txt file with the current package versions. Could you write down the versions of all imported python modules you used?

**Comment by ChiCheng45** [comment]
> Nice work! Since the project got a little bit more complicated, we should compile a requirements txt file with the current package versions. Could you write down the versions of all imported python modules you used?

I've added a requirements.txt generated using pycharm with the latest commit, I didn't specify version since I'd imagine it should work fine for most versions. But here are the versions that I used.

pandas: 1.1.5
numpy: 1.19.1
scipy: 1.5.2
setuptools: 51.0.0.post20201207
cython: 0.29.21

**Comment by oziogos** [comment]
> > Nice work! Since the project got a little bit more complicated, we should compile a requirements txt file with the current package versions. Could you write down the versions of all imported python modules you used?
> 
> I've added a requirements.txt generated using pycharm with the latest commit, I didn't specify version since I'd imagine it should work fine for most versions. But here are the versions that I used.
> 
> pandas: 1.1.5
> numpy: 1.19.1
> scipy: 1.5.2
> setuptools: 51.0.0.post20201207
> cython: 0.29.21

Great! My versions are older than yours, but everything works fine. So I'll put my values in the file.

**Pull request merged**